Sources/PackageModel/ResolvedModels.swift

```diff
@@ -178,6 +178,11 @@ public final class ResolvedProduct: ObjectIdentifierProtocol, CustomStringConver
     public var description: String {
         return "<ResolvedProduct: \(name)>"
     }
+
+    /// True if this product contains Swift targets.
+    public var containsSwiftTargets: Bool {
+      return targets.contains { $0.underlyingTarget is SwiftTarget }
```

Comment:
It looks like this is just top-level targets; should it search for any targets?

Reply:
Yeah, you probably want to flatMap the `recursiveDependencies` first. Something like:

targets.flatMap { $0.recursiveDependencies }.contains { $0.underlyingTarget is SwiftTarget }

Reply:
Not sure if `target.lazy.flatMap` will help with performance here?

Reply:
There's no reason to build a temporary here, so lazy would work, or `targets.contains { $0.recursiveDependencies.contains { ... } }`

Reply:
One complications here is that we should be stopping this search at the first dynamic product in the dependency graph because the top-level product could be just made up of C targets and getting Swift targets via a dylib.

Reply:
One thing though is that C targets can't import Swift targets in SwiftPM (at least not right now), so we can just do this for the top-level targets (as you've already done) and handle the other complications when we do start supporting the other case. Can you add a comment and file a JIRA for that?

Reply:
https://bugs.swift.org/browse/SR-10791 Thanks!